Source/ARTJsonLikeEncoder.m

```diff
@@ -645,11 +651,31 @@ - (ARTDeviceDetails *)deviceDetailsFromDictionary:(NSDictionary *)input error:(N
     deviceDetails.formFactor = [input artString:@"formFactor"];
     deviceDetails.metadata = [input valueForKey:@"metadata"];
     deviceDetails.push = [self devicePushDetailsFromDictionary:input[@"push"] error:error];
-    deviceDetails.updateToken = [input artString:@"updateToken"];
 
     return deviceDetails;
 }
 
+- (ARTDeviceIdentityTokenDetails *)deviceIdentityTokenDetailsFromDictionary:(NSDictionary *)input error:(NSError * __autoreleasing *)error {
+    [_logger verbose:@"RS:%p ARTJsonLikeEncoder<%@>: deviceIdentityTokenDetailsFromDictionary %@", _rest, [_delegate formatAsString], input];
+
+    if (![input isKindOfClass:[NSDictionary class]]) {
```

Comment:
Well, I did that way because it's how the "others do". Maybe it's better to open an issue with that concern and update all the `*FromDictionary:` methods. Thanks.